app/dao/services_dao.py

```diff
@@ -194,16 +195,30 @@ def dao_fetch_service_by_inbound_number(number):
 
 
 def dao_fetch_service_by_id_with_api_keys(service_id, only_active=False):
-    query = Service.query.filter_by(
+    # use read-db engine bound to the read intance instead of the default
+    # engine bound to the write instance of our cluster
+    reader = db.engines['read-db']
+    session = scoped_session(sessionmaker(bind=reader))
+    current_app.logger.info(f"dao_fetch_service_by_id_with_api_keys read engine is {reader}")
+
+    query = session.query(Service).filter_by(
         id=service_id
     ).options(
         joinedload('api_keys')
     )
+    current_app.logger.info(f"dao_fetch_service_by_id_with_api_keys query is {query}")
 
     if only_active:
         query = query.filter(Service.active)
 
-    return query.one()
+    session.close()
+    result = query.one()
```

Comment:
See [here](https://docs.sqlalchemy.org/en/14/orm/session_basics.html#opening-and-closing-a-session).  The would probably work better as a context block.  That would eliminate the explicit "close".  Something like:
```python
with scoped_session(...) as session:
  ...
  result = query.one()
```

Reply:
done